src/components/footer/config.ts

```diff
@@ -0,0 +1,333 @@
+            fr: 'TODO',
+            it: 'TODO',
+          },
+        },
+        {
+          translationKey: 'footer.sections.brand.insurance',
+          visibilitySettings: {
+            brand: { as24: true, ms24: false },
+          },
+          link: {
+            de: 'https://www.financescout24.ch/de/autoversicherung-finden?utm_source=autoscout24.ch&utm_medium=web&utm_campaign=link_footer_car_',
+            en: 'https://www.financescout24.ch/de/autoversicherung-finden?utm_source=autoscout24.ch&utm_medium=web&utm_campaign=link_footer_car_',
+            fr: 'TODO',
+            it: 'TODO',
+          },
+          target: '_blank',
+        },
+      ],
+    },
+  ],
+  apps: {
+    android: [],
+    apple: [],
+  },
+  socialMedia: {
+    facebook: [],
+    instagram: [],
+    twitter: [],
+    youtube: [],
+  },
+  companies: [],
+};
+
+export class FooterConfig {
+  brand: Brand;
+  config: FooterConfigInterface;
+  mappedConfig: null | FooterConfigInstance;
+  isProduction: boolean;
+  useAbsoluteUrls: boolean;
+
+  constructor({
+    config,
+    brand,
+    isProduction = false, // TODO: switch default before merge
+    useAbsoluteUrls = true, // TODO: switch default before merge
+  }: {
+    config: FooterConfigInterface;
+    brand: Brand;
+    isProduction?: boolean;
+    useAbsoluteUrls?: boolean;
+  }) {
+    this.brand = brand;
+    this.config = config;
+    this.isProduction = isProduction;
+    this.useAbsoluteUrls = useAbsoluteUrls;
+
+    this.mappedConfig = null;
+  }
+
+  getMappedConfig(): FooterConfigInstance {
+    if (this.mappedConfig) return this.mappedConfig;
+
+    const mappedConfig = {
+      sections: this.config.sections.map(this.mapSectionConfig.bind(this)),
+      apps: {
+        android: this.mapLinkArray(this.config.apps.android),
+        apple: this.mapLinkArray(this.config.apps.apple),
+      },
+      socialMedia: {
+        facebook: this.mapLinkArray(this.config.socialMedia.facebook),
+        instagram: this.mapLinkArray(this.config.socialMedia.instagram),
+        twitter: this.mapLinkArray(this.config.socialMedia.twitter),
+        youtube: this.mapLinkArray(this.config.socialMedia.youtube),
+      },
+      companies: this.mapLinkArray(this.config.companies),
+    };
+    this.mappedConfig = mappedConfig;
+    return this.mappedConfig;
+  }
+
+  private mapSectionConfig(section: LinkSectionConfig) {
+    const mappedSection = {
+      title: this.mapLinkArray(section.title),
+      items: this.mapLinkArray(section.items),
+    };
+
+    return mappedSection;
+  }
+
+  private mapLinkArray(linkConfigArray: LinkConfig[]): LinkInstance[] {
+    return linkConfigArray.map((linkConfig) => this.mapLink(linkConfig));
+  }
+
+  private mapLink(config: LinkConfig): LinkInstance {
+    return new Link({
+      config,
+      brand: this.brand,
+      isProduction: this.isProduction,
+      useAbsoluteUrls: this.useAbsoluteUrls,
+    });
+  }
+}
+
+// !!CMP Link
+class Link {
```

Comment:
maybe extract Link to a separate file along with the interfaces that belong to it. I find it hard to jump back and forth between the class and the types